### ISSUE TITLE
Could you please share some details about the Medical Performance?

### ISSUE THREAD

**Issue by s1ghhh** [OptimalScale/LMFlow#548]
Thank you very much for open-sourcing such an excellent project. I am very interested in the experimental results of your [Medical Performance](https://github.com/OptimalScale/LMFlow#medical-performance), specifically the table.  

Could you please share the specific hyperparameters for Task-tuned 7b/13b? It would be very helpful for me. 

Additionally, I'm also interested in how to evaluate the performance of ChatGPT. Could you share the prompts used during testing? I would also like to know how to perform out-of-domain testing with ChatGPT. Intuitively, I'm unable to fine-tune ChatGPT, which makes it difficult to differentiate between in-domain and out-of-domain. 

Lastly, I believe there might be some minor issues in the table. [The "Human (pass)" percentage for MedQA-USMLE (OOD) should be 60%, and for MedMCQA (ID) it should be 50%](https://arxiv.org/abs/2207.08143). 

Thank you for your contributions to the community, and I look forward to your response.

**Comment by shizhediao** [comment]
Hi，
Please refer to these two scripts: https://github.com/OptimalScale/LMFlow/blob/main/scripts/run_finetune_with_lora.sh and https://github.com/OptimalScale/LMFlow/blob/main/scripts/run_finetune.sh.
You might want to set num_epochs to 3 and try 1e-4, 2e-5 as the learning rate.
Here are the prompts for ChatGPT.

```
if args.dataset == "pubmedqa":
      with open(args.dataset_path, encoding='utf-8') as f:
            json_data = json.load(f)
            for line in json_data:
                #tokens = tokenizer.tokenize(line["context"])
                tokens_not_tokenize = line["context"]
                q = "Context: " + line["context"] + "\n Question: " +  line["question"].strip() + "\nA) yes\nB) no\nC) maybe"
                questions.append(q)
                answers.append(line["final_decision"])
    elif args.dataset == "medmcqa":
      with open(args.dataset_path, encoding='utf-8') as f:
            json_data = json.load(f)
            for line in json_data:
                q = "Question: " + line["question"] + " (A) " + line["opa"] + \
                        " (B) " + line["opb"] +  " (C) " + line["opc"] +  " (D) " + line["opd"] + "."
                questions.append(q)
                answers.append(line["answer"])  
    elif args.dataset == "usmle":
      with open(args.dataset_path, encoding='utf-8') as f:
            json_data = json.load(f)
            for line in json_data:
                q = "Question: " + line["question"] + " (A) " + line["opa"] + \
                        " (B) " + line["opb"] +  " (C) " + line["opc"] +  " (D) " + line["opd"] + "."
                questions.append(q)
                answers.append(line["answer"])
```


As for the pass rate and expert rate, thank you so much for pointing out! We have

**Comment by s1ghhh** [comment]
> Hi， Please refer to these two scripts: https://github.com/OptimalScale/LMFlow/blob/main/scripts/run_finetune_with_lora.sh and https://github.com/OptimalScale/LMFlow/blob/main/scripts/run_finetune.sh. You might want to set num_epochs to 3 and try 1e-4, 2e-5 as the learning rate. Here are the prompts for ChatGPT. The ID and OOD are mainly designed for other models instead of ChatGPT.
> 
> ```
>     if args.dataset == "pubmedqa":
>       with open(args.dataset_path, encoding='utf-8') as f:
>             json_data = json.load(f)
>             for line in json_data:
>                 #tokens = tokenizer.tokenize(line["context"])
>                 tokens_not_tokenize = line["context"]
>                 q = "Context: " + line["context"] + "\n Question: " +  line["question"].strip() + "\nA) yes\nB) no\nC) maybe"
>                 questions.append(q)
>                 answers.append(line["final_decision"])
>     elif args.dataset == "medmcqa":
>       with open(args.dataset_path, encoding='utf-8') as f:
>             json_data = json.load(f)
>             for line in json_data:
>                 q = "Question: " + line["question"] + " (A) " + line["opa"] + \
>                         " (B) " + line["opb"] +  " (C) " + line["opc"] +  " (D) " + line["opd"] + "."
>                 questions.append(q)
>                 answers.append(line["answer"])  
>     elif args.dataset == "usmle":
>       with open(args.dataset_path, encoding='utf-8') as f:
>             json_data = json.load(f)
>             for line in json_data:
>                 q = "Question: " + line["question"] + " (A) " + line["opa"] + \
>                         " (B) " + line["opb"] +  " (C) " + line["opc"] +  " (D) " + line["opd"] + "."
>                 questions.append(q)
>                 answers.append(line["answer"])
> ```
> 
> ```
>     if args.dataset == "pubmedqa":
>         args.dataset_path = "./dataset/medical_dataset/PubMedQA/pqa_labeled_1000.json"
>         args.direct_answer_trigger = "\nTherefore, among A through C, the answer is"
>         args.dataset_size = 1000    
>     elif args.dataset == "medmcqa":
>         args.dataset_path = "./dataset/medical_dataset/MedMCQA/medmcqa_valid_4183.json"
>         args.direct_answer_trigger = "\nTherefore, the answer (A or B or C or D) is"
>         args.dataset_size = 4183  
>     elif args.dataset == "usmle":
>         args.dataset_path = "./dataset/medical_dataset/MedQA-USMLE/usmle_test_1273.json"
>         args.direct_answer_trigger = "\nTherefore, the answer (A or B or C or D) is"
>         args.dataset_size = 1273
> ```
> 
> As for the pass rate and expert rate, thank you so much for pointing out! We have

Thank you for your selfless sharing. I'll try to reproduce and report back when I get results. 🙏

**Comment by shizhediao** [comment]
Hi,
We do not use `examples/benchmarking.py`. Basically, it just queries the openai API with the question and the prompt structure is shown here.

> Hi， Please refer to these two scripts: https://github.com/OptimalScale/LMFlow/blob/main/scripts/run_finetune_with_lora.sh and https://github.com/OptimalScale/LMFlow/blob/main/scripts/run_finetune.sh. You might want to set num_epochs to 3 and try 1e-4, 2e-5 as the learning rate. Here are the prompts for ChatGPT. The ID and OOD are mainly designed for other models instead of ChatGPT.
> 
> ```
>     if args.dataset == "pubmedqa":
>       with open(args.dataset_path, encoding='utf-8') as f:
>             json_data = json.load(f)
>             for line in json_data:
>                 #tokens = tokenizer.tokenize(line["context"])
>                 tokens_not_tokenize = line["context"]
>                 q = "Context: " + line["context"] + "\n Question: " +  line["question"].strip() + "\nA) yes\nB) no\nC) maybe"
>                 questions.append(q)
>                 answers.append(line["final_decision"])
>     elif args.dataset == "medmcqa":
>       with open(args.dataset_path, encoding='utf-8') as f:
>             json_data = json.load(f)
>             for line in json_data:
>                 q = "Question: " + line["question"] + " (A) " + line["opa"] + \
>                         " (B) " + line["opb"] +  " (C) " + line["opc"] +  " (D) " + line["opd"] + "."
>                 questions.append(q)
>                 answers.append(line["answer"])  
>     elif args.dataset == "usmle":
>       with open(args.dataset_path, encoding='utf-8') as f:
>             json_data = json.load(f)
>             for line in json_data:
>                 q = "Question: " + line["question"] + " (A) " + line["opa"] + \
>                         " (B) " + line["opb"] +  " (C) " + line["opc"] +  " (D) " + line["opd"] + "."
>                 questions.append(q)
>                 answers.append(line["answer"])
> ```
> 
> ```
>     if args.dataset == "pubmedqa":
>         args.dataset_path = "./dataset/medical_dataset/PubMedQA/pqa_labeled_1000.json"
>         args.direct_answer_trigger = "\nTherefore, among A through C, the answer is"
>         args.dataset_size = 1000    
>     elif args.dataset == "medmcqa":
>         args.dataset_path = "./dataset/medical_dataset/MedMCQA/medmcqa_valid_4183.json"
>         args.direct_answer_trigger = "\nTherefore, the answer (A or B or C or D) is"
>         args.dataset_size = 4183  
>     elif args.dataset == "usmle":
>         args.dataset_path = "./dataset/medical_dataset/MedQA-USMLE/usmle_test_1273.json"
>         args.direct_answer_trigger = "\nTherefore, the answer (A or B or C or D) is"
>         args.dataset_size = 1273
> ```
> 
> As for the pass rate and expert rate, thank you so much for pointing out! We have

**Comment by shizhediao** [comment]
This issue has been marked as stale because it has not had recent activity. If you think this still needs to be addressed please feel free to reopen this issue. Thanks

**Comment by eshaanagarwal** [comment]
Hey can anybody share the script for Medical Performannce benchmark for gpt3.5 and 4 ? I was facing some issues regarding regex and extraction of answer.
